### ISSUE TITLE
Error: configuration for aws_alb_listener_rule.alb_store_rule_blockcarbon still contains unknown values during apply (this is a bug in Terraform; please report it!)

### ISSUE THREAD

**Issue by madcowthafirst** [hashicorp/terraform#24443]
### Terraform Version

```
Terraform v0.12.24
```

### Terraform Configuration Files

```hcl

FILE main.tf

module "lb-listener-ws-store" {
  source = "./modules/lb_listener_ssl"

  newvpc_lb_arn                     = module.alb-wsexternal.alb_arn
  newvpc_lb_listener_certificate_id = data.aws_acm_certificate.alb-ws-store-ext-cert.arn
  newvpc_lb_listener_cfg            = var.newvpc_alb_ws_store_external_listener_cfg
  newvpc_lb_tg_ids                  = module.lb-tg-ws-store.tg_ids
}

resource "aws_alb_listener_rule" "alb_store_rule_blockcarbon" {

  listener_arn = module.lb-listener-ws-store[1].arn
  priority = "10"


  action {
    type = "fixed-response"

    fixed_response {
      content_type = "text/plain"
      message_body = "Access Denied"
      status_code  = "401"
    }
  }

  condition {
    path_pattern {
      values = ["/carbon*"]
    }
  }
}

FILE modules/lb_listener_ssl/main.tf

variable "newvpc_lb_arn"                        {}
variable "newvpc_lb_listener_certificate_id"    {}
variable "newvpc_lb_listener_cfg"               {}
variable "newvpc_lb_tg_ids"                     {}

resource "aws_lb_listener" "listener" {
  count                 = length(var.newvpc_lb_listener_cfg)

  load_balancer_arn     = var.newvpc_lb_arn
  port                  = lookup(var.newvpc_lb_listener_cfg[count.index], "port")
  protocol              = lookup(var.newvpc_lb_listener_cfg[count.index], "protocol")
  ssl_policy            = lookup(var.newvpc_lb_listener_cfg[count.index], "ssl_policy")
  certificate_arn       = var.newvpc_lb_listener_certificate_id

  default_action {
    type             = "forward"
    target_group_arn = var.newvpc_lb_tg_ids[count.index]
  }
}

FILE modules/lb_listener_ssl/output.tf

output "arn" {
  value       = aws_lb_listener.listener.*.arn
  description = "The id of the listener"
}



```

### Debug Output

Please find attached.

### Crash Output

n.a. 

### Expected Behavior

The ALB rule should be created for the linked application load balancer.

### Actual Behavior

Using the following terraform code results in an error when running terraform apply.
Terraform plan finishes successfully but terraform apply bails out with the following error.

"Error: configuration for aws_alb_listener_rule.alb_store_rule_blockcarbon still contains unknown values during apply (this is a bug in Terraform; please report it!)"

### Steps to Reproduce

terraform apply -var-file=./env-to-use/vars.tfvars

### Additional Context

n.a.

### References

Nope

[terraform_output.log](https://github.com/hashicorp/terraform/files/4375545/terraform_output.log)

**Comment by danieldreier** [comment]
@madcowthafirst thanks for reporting this, and I apologize for how long it's taken me to respond. I think that this reproduction case is a little incomplete, so I've copied it into a [GitHub repo I recently started using to collaborate with folks on issue reproduction cases](https://github.com/danieldreier/terraform-issue-reproductions/tree/master/24443). It'll be way easier to work on it together in GitHub than inline text here.

I'm hesitant to start inventing details to fill this in, because it's (a) a ton of work, and (b) often when I've done that in the past, it turned out that a detail I invented mattered, and then the reporter had to re-do it all anyway.

Can you please try and run that reproduction case, and fill it in to the point where it actually runs and reproduces the issue you reported? I need a full, runnable reproduction case in order to be able to put this in front of an engineer. I really appreciate you reporting this!

**Comment by madcowthafirst** [comment]
@danieldreier Hi Daniel,

Thanks for your follow up. Can you help me a bit how to submit the reproduction case to github?
I was able to create a standalone reproduction case for this issue.

I did a git clone of the terraform-issue-reproductions repo but I don't know how I need to push it?
Just trying to push it to a new branch? But probably I have to take another route to get it in this github repo.

Thanks in advance.

**Comment by madcowthafirst** [comment]
Created a pull request to merge the code with a working reprodution case.

https://github.com/danieldreier/terraform-issue-reproductions/pull/7

**Comment by mytuanZ** [comment]
Hello,
Yesterday I created a security group module and got the same error message
---
name: Bug report
about: Cannot create resource aws_security_group, still contains unknown values during apply. 
labels: bug, new

---

### Terraform Version


```
Terraform v0.12.25
```

### Terraform Configuration Files


```
#This is the main.tf file, include all modules

terraform {
  required_version = ">= 0.12.3"
  required_providers {
    aws = "= 3.0.0"
  }
}
provider "aws" {
  region  = var.region
  shared_credentials_file = ".aws/credentials"
}

module "aws_vpc" {
  create_vpc  = var.create_vpc
  source      = "../modules/vpc"

  vpc_cidr_block                     = var.vpc_cidr_block
  vpc_instance_tenancy               = var.vpc_instance_tenancy
  vpc_enable_dns_support             = var.vpc_enable_dns_support
  vpc_enable_dns_hostnames           = var.vpc_enable_dns_hostnames
  vpc_enable_classiclink             = var.vpc_enable_classiclink
  vpc_enable_classiclink_dns_support = var.vpc_enable_classiclink_dns_support
  vpc_enable_ipv6                    = var.vpc_enable_ipv6

  vpc_tag_name  = var.vpc_tag_name
  environment   = var.environment
  project       = var.project
  }

module "aws_security_group_test" {
  source = "../modules/security_groups"

  vpc_id                      = module.aws_vpc[0].vpc_id
  security_group_name         = var.security_group_name
  security_group_description  = var.security_group_description
  security_group_tag_name     = var.security_group_tag_name
}  
```

```
#This is the main variables.tf file.

variable "region" {
  default = "ap-southeast-1"
}
#
# varialbes for resource VPC
#
variable "create_vpc" {
  description = "Controls if VPC should be created (it affects almost all resources)"
  type        = bool
  default     = true
}

variable "vpc_cidr_block" {
  default = "10.200.96.0/22"
}
variable "vpc_instance_tenancy" {
  default = "default"
}

variable "vpc_enable_dns_support" {
  default = "true"
}

variable "vpc_enable_dns_hostnames" {
  default = "true"
}

variable "vpc_enable_classiclink" {
  default = "false"
}

variable "vpc_enable_classiclink_dns_support" {
  default = "false"
}
variable "vpc_enable_ipv6" {
  default = "false"
}

variable "vpc_tag_name" {
  type        = string
  default     = "tf_wso2"
}

variable "environment" {
  type        = string
  default     = "dev"
}

variable "project" {
  type        = string
  default     = "integration"
}

#
#Security Group
#
variable "security_group_name" {
  type = string
  default = "test_tf_wso2_sg"
}
variable "security_group_description" {
  type = string
  default = "test_tf_wso2_sg"

}
variable "security_group_tag_name" {
  type = string
  default = "test_tf_wso2_sg"
}
```

```
#This is the vpc module ../modules/vpc/main.tf
 
resource "aws_vpc" "DEV_VPC" {
  count = var.create_vpc ? 1 : 0

  cidr_block           = var.vpc_cidr_block
  instance_tenancy     = var.vpc_instance_tenancy
  enable_dns_support   = var.vpc_enable_dns_support
  enable_dns_hostnames = var.vpc_enable_dns_hostnames
  enable_classiclink               = var.vpc_enable_classiclink
  enable_classiclink_dns_support   = var.vpc_enable_classiclink_dns_support
  assign_generated_ipv6_cidr_block = var.vpc_enable_ipv6

  tags = {
    "Name"        = var.vpc_tag_name,
    "Environment" = var.environment,
    "Project"     = var.project
  }
}
```

```
#This is the variables for vpc module  ../modules/vpc/variables.tf

variable "create_vpc" {
  description = "Controls if VPC should be created (it affects almost all resources)"
  type        = bool
  default     = true
}
variable "vpc_cidr_block" {}
variable "vpc_instance_tenancy" {}
variable "vpc_enable_dns_support" {}
variable "vpc_enable_dns_hostnames" {}
variable "vpc_enable_classiclink" {}
variable "vpc_enable_classiclink_dns_support" {}
variable "vpc_enable_ipv6" {}
variable "vpc_tag_name" {
  description = "Name to be used on all the resources as identifier"
  type        = string
}

variable "environment" {
  description = "Name to be used on all the resources as identifier"
  type        = string
}

variable "project" {
  description = "Name to be used on all the resources as identifier"
  type        = string
}
```

```
#This is the outputs.tf in vpc module ../modules/vpc/outputs.tf

output "vpc_id" {
  description = "The ID of the VPC"
  value       = aws_vpc.DEV_VPC.*.id
}
```

```
#This is the main.tf in security_groups module ../modules/security_groups/main.tf

resource "aws_security_group" "sec_group" {
  name        = var.security_group_name
  description = var.security_group_description
  vpc_id      = var.vpc_id
  tags = {
    Name = var.security_group_tag_name
  }
}
```

```
#This is the variables for security_groups module  ../modules/security_groups/variables.tf

variable "vpc_id" {
  type = string
}
variable "security_group_name" {
  type = string
}
variable "security_group_description" {
  type = string
}
variable "security_group_tag_name" {
  type = string
}
``` 

### Debug Output


### Crash Output


### Expected Behavior

A security group should be created.
### Actual Behavior

I got the error message:
```
Error: configuration for module.aws_security_group_test.aws_security_group.security_group still contains unknown values during apply (this is a bug in Terraform; please report it!)
```
### Steps to Reproduce

1. `terraform init`
2. `terraform plan`
2. `terraform apply -auto-approve`

### Additional Context


### References


If I update it to v0.13.0, It fixed.
Thanks

**Comment by mildwonkey** [comment]
Hi @madcowthafirst !

I see from your last comment that you're already aware that this is fixed in Terraform v0.13: 
If I update it to v0.13.0, it can create security group.

Based on that, I am going to close this issue -thank you for the update, and I'm sorry for the problem!

**Comment by madcowthafirst** [comment]
Just was able to test it today with terraform version 13.2 and noticed that it now works! Thanks for getting this fixed.

Cheers Marco!
